func/internal/podevaluator.go

```diff
@@ -597,9 +619,70 @@ func (pm *podManager) getCustomAuth(ref name.Reference, registryAuthSecretPath s
 	return authn.FromConfig(dockerConfig.Auths[ref.Context().RegistryStr()]), nil
 }
 
+func loadTLSConfig(caCertPath string) (*tls.Config, error) {
+	// Read the CA certificate file
+	caCert, err := os.ReadFile(caCertPath)
+	if err != nil {
+		return nil, err
+	}
+	// Append the CA certificate to the system pool
+	caCertPool := x509.NewCertPool()
+	if !caCertPool.AppendCertsFromPEM(caCert) {
+		return nil, err
+	}
+	// Create a tls.Config with the CA pool
+	tlsConfig := &tls.Config{
+		RootCAs: caCertPool,
+	}
+	return tlsConfig, nil
+}
+
+func createTransport(tlsConfig *tls.Config) *http.Transport {
+	return &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+}
+
+func getImage(ctx context.Context, ref name.Reference, auth authn.Authenticator, image string, enablePrivateRegistries bool, enableTlsRegistries bool, tlsSecretPath string) (v1.Image, error) {
+	if enablePrivateRegistries && !strings.HasPrefix(image, defaultRegistry) && enableTlsRegistries {
+		tlsFile := "ca.crt"
+		// Check if mounted secret location contains CA file.
+		_, err := os.Stat(tlsSecretPath)
+		if os.IsNotExist(err) {
+			return nil, err
+		}
+		if _, err := os.Stat(filepath.Join(tlsSecretPath, "ca.crt")); os.IsNotExist(err) {
+			klog.Error(err)
```

Comment:
Should this log be inside the check-for-ca.pem if block, to indicate that we couldn't find a `ca` file with either extension?
(or perhaps make this one a Warn and add an Error in the .pem check?)

Reply:
Updated code looks good - OK to close